pandas/core/missing.py

```diff
@@ -39,6 +40,21 @@
     from pandas import Index
 
 
+def check_value_size(value, mask: np.ndarray, length: int):
```

Comment:
as this is also masking the values, a different name may be more appropriate.

or maybe return none. just do the validation here and just repeat the masking in the array functions, so it it is clear to readers what is going on.

Reply:
> just do the validation here and just repeat the masking in the array functions, so it it is clear to readers what is going on.

thats not a bad idea.  i think we do something similar in array_algos.putmask